### PR TITLE
[MODULAR] [QOL] Bitrunner Alt-titles + Others + QoL

### PR DESCRIPTION
**Pull request by W4rd3nn**
## About The Pull Request
Kind of a big one, but it only affects one file so I don't see the need to split it up.

This does a few things

- Adds Bitrunner alt-titles
- Alphabetizes alt-titles
- Adds junior/newbie titles where appropriate
- Adds some misc new titles

<details>
<summary>Changed Titles</summary>

- ADD: BARBER **Aethestician**
- ADD: BITRUNNER **Bitdomain Technician**
- ADD: BITRUNNER **Data Retrieval Specialist**
- ADD: BITRUNNER **Netdiver**
- ADD: BITRUNNER **Pod Jockey**
- ADD: BITRUNNER **Union Bitrunner**
- ADD: BITRUNNER **Junior Runner**
- ADD: BOTANIST **Mycologist**
- ADD: BOTANIST **Junior Botanist**
- ADD: CARGO TECHNICIAN **Commodities Trader**
- ADD: CARGO TECHNICIAN **Receiving Clerk**
- ADD: CORONER **Forensic Pathologist**
- ADD: CORONER **Medical Examiner**
- ADD: CURATOR **Conservator**
- ADD: CYBORG **Machine**
- ADD: DOCTOR **Medical Student**
- ADD: GENETICIST **Biologist**
- ADD: MIME **Mummer**
- ADD: NANOTRASEN CONSULTANT **Nanotrasen Advisor**
- ADD: ROBOTICIST **Machinist**
- ADD: SHAFT MINER **Apprentice Miner**
- ADD: VIROLOGIST **Epidemiologist**
- CHANGE: LAWYER **"Defence Attorney"** > **"Defense Attorney"** (go away british)
- REMOVE: STATION ENGINEER **Engineering Trainee** (We already have a newbie title for engineers)

  
</details>

- Alphabetization is only broken in 3 key areas. The base title is always placed first. The junior title is always placed last. The minimum, maximum, supermax, protective custody prisoners are in order of security, not alphabetization. Else it looks weird.
- Engineering Trainee was removed because Apprentice Engineer already exists.
- Roles with only one alt titles were given a second excluding the department guards
- Barber was given an alt-title befitting a makeup artist
- Botanist was given Mycologist (since there are mushooms in the role) and Junior Botanist for newbie botankees.
- Cargo tech was given two new alt-titles to fit duties they have. Namely checking manifests and trading minerals.
- Coroner titles were a bit anemic, so it was given two more.
- Curator was given a new title because i feel like it
- Cyborg was given Machine, a level ordered below Robot. (good for roombas and non-intelligent borgs)
- Med Doc was given a newbie title
- Geneticist was given a new title since it only had one alt title and it was lame
- Mime was given a new title since it only had one alt title and it was lame
- Nanotrasen Consultant was given a new title since it only had one alt title and i felt like it
- Roboticist was given a new title to represent general knowledge of the department
- Shaft miner was given a newbie title
- Virologist was given a new title since it only had one alt title and it was okay.
- Defence Attorney to Defense Attorney. We aren't in BRIT LAND are we?

## How This Contributes To The Skyrat Roleplay Experience
Alt titles are a big part of life here on skyrat. Roles who dont have any or who have only a few should get more options. It also allows player to "specialize" in their department ICly. Makes things nice and epic and cool. Also the alphabetization refactor will make finding titles at a glance much easier.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

image

image

image

</details>

## Changelog

:cl:Motho
add: Bitrunners can now have alternative job titles. FTU urges that these titles are purely cosmetic and not representative of bitrunning ability.
add: Barbers, Botanists, Warehouse Techs, Coroners, Curators, Cyborgs, Geneticists, Mimes, Nanotrasen Consultants, Roboticists, and Virologists enjoy new alternative job titles.
add: Certain jobs now have Trainee/Newbie alternative job titles ordered at the very bottom of the title selection dialog. If you or your character are new to the job/department, set your title so your colleagues are aware! 
del: Removed Engineering Trainee.
qol: Alphabetized alt-titles excluding two key areas. The base title, and the newbie title.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (Stalkeros2) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:60]: (Cybersportsman or someshit would be really funny/fitting tbh.)

**Comment by nikothedude** [comment]
This is not a refactor, not a code improvement either, you'd honestly want to replace that with spellcheck or smthn

A refactor is a moderate or large redesign of a system without changing its function, this is neither

**Comment by W4rd3nn** [comment]
changed to QoL since its closer.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (ghost) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:63]: Honestly I'm not really sold on the Bitdomain Technician. It's really mouthy and doesn't really fit the role either - they're not technicians, they don't maintain the servers. They're raiding virtual servers (honestly from a lore standpoint IDK if this is even legal now that I think about it...)

Same thoughts on Pod Jockey but for different reasons - while it's fitting it just sounds a bit gross/unprofessional. There's no reason an official station ID would call them it

Also Junior Runner needs to be Junior Bitrunner (bc normal running isn't what they do lol)

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (ghost) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: ~~Is this not in use by science? It seems more like a Cytology-rename than genetics...~~
It's technically not in use, but it's really similar to the Xenobiologist. Also doesn't really cover the scope of the job, which works in genetics and DNA, not biology...
If you need a replacement idea you could use this one instead; simple combination of both the current ones
```suggestion
		"Genetics Researcher",
```

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (ghost) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:280]: > an actor in a traditional [masked](https://www.google.com/search?client=firefox-b-1-d&sca_esv=570352775&channel=fenc&sxsrf=AM9HkKlBUFTzJz2CdPV5TecR-ll2YQOuug:1696372563399&q=masked&si=ALGXSlbD4fKmSL7CRU364kGH2u8ksgLkZxtvHUZalAD46f29zjXrd7wx8LgFWBrQV2pXANbFOga8MH6VaoDmB5spZthPD6Mjfg%3D%3D&expnd=1) [mime](https://www.google.com/search?client=firefox-b-1-d&sca_esv=570352775&channel=fenc&sxsrf=AM9HkKlBUFTzJz2CdPV5TecR-ll2YQOuug:1696372563399&q=mime&si=ALGXSlYftE7XViDpYdZ4PMnebYlvxQe7U2vSwd_bwbBNYyaZY8_bv--UH-FQqteaPrqmo8tsaksLN8z4Rot-qD0s-_Di9jCKDg%3D%3D&expnd=1), especially of a type associated with Christmas and popular in England in the 18th and early 19th centuries.
> 
>     a pantomimist.

    
   
Huh. Neat. Today I Learned.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (ghost) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: I don't actually have the authority for this one but I'd like a maintainer's input - can these actually be removed? Having the prisoners not all labelled consistently can _and has_ made it hard for security to find them in records, as the primary way of finding them is searching by job title.

**Review by ghost — COMMENTED** [review]
Mostly just personal opinion but it doesn't hurt to share those thoughts. Some are (in my opinion of course) very valid. I.E. the Junior Runner needing to be a Junior **Bit**runner

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (W4rd3nn) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:63]: This was a headmin suggestion so i'll leave it up to dem. Pod Jockey is poggers tho u cant tell me what 2 do >:(

Comment (ghost) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:63]: Eugh! Ech! Ptooey!!

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (W4rd3nn) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:60]: goof is apparently gonna bug fix my PR after merge to add Gamer.............

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (SpaceLoveSs13) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:66]: Instead of Netdiver, how about Netrunner?
Also Pod Jockey is way too much of an unprofessional and irrelevant name for a role like bitrunner in my opinion.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (SpaceLoveSs13) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: This is an odd choice. Calling a somewhat "sentient" piece of moving, talking robot a mere "machine". That just larps it with every other machine on the station. Not really a fan of this.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (SpaceLoveSs13) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:225]: We already have Medical resident which is a junior intern-like role for MDs. The same reason why you removed Engineering trainee and kept Apprentice Engineering basically.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (nikothedude) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: Plenty of synth players, myself included, play synths that cant be called sentient or sapient, or would perfer to at least pretend its the case

Comment (ghost) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: I think that Machine is better when I can use it as an insult and it's not actually just the truth :trollface:

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (W4rd3nn) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:66]: Netrunner is a copyright of R. Talsorian Games Inc.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (W4rd3nn) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:225]: A Resident is expected to know their job pretty well. Residency, to me, is just for RP flavor. The newbie titles are meant to CLEARLY outline who is actually, physically, new to the job so they can be focused on for teaching. Hence med student.

**Comment by nikothedude** [comment]
of all the prs to get noticable reviews i didnt expect it to be thos one

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (SpaceLoveSs13) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:66]: That is true, my bad. But what about Pod Jockey?

Comment (W4rd3nn) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm:66]: pod jockey is poggers. i think bitrunners, given the nature of their jobs, have a bit of leeway in their titles. I mean, their official title is bitrunner. Sounds like some made up video game shit. That's why I put Data Retrieval Specialist in there as a more corporate-lingo themed title.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: I'm not fond of having "Machine" be a choice here, that isn't accurate as to what their function is. They're not a stationary object, they're moving around and performing duties.
```suggestion
```

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (GoldenAlpharex) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: I agree with Orion on this one.

File: modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm
Comment (GoldenAlpharex) [reply modular_skyrat/modules/alternative_job_titles/code/alt_job_titles.dm]: Yeah, I think that's a fair point. It's not like if it made much sense for them to not be called directly "Prisoner" in some capacity, since it's meant to be something that you're not in of your own accord, really.

**Review by GoldenAlpharex — DISMISSED** [review]

**Comment by W4rd3nn** [comment]
All reviews resolved............

**Review by vinylspiders — APPROVED** [review]

**Pull request merged**
